cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -57,10 +58,15 @@ func CRDStatusToNCRequest(crdStatus nnc.NodeNetworkConfigStatus) (cns.CreateNetw
 				return ncRequest, fmt.Errorf("Invalid SecondaryIP %s:", ipAssignment.IP)
 			}
 
-			secondaryIPConfig = cns.SecondaryIPConfig{
-				IPAddress: ip.String(),
+			ipConfig, ok := ncRequest.SecondaryIPConfigs[ipAssignment.Name]
+			if !ok || (ok && ipConfig.IPAddress != ip.String()) {
+				ncVersion, _ := strconv.Atoi(ncRequest.Version)
```

Comment:
We dont need this converstion to happen inside a loop

Reply:
We need this logic. It's the same question as your second comment. Please see explanation there.